Windows Rust Slave Fixes

jacderida opened the pull request:
Hi Stephen,

This introduces another VM for building Rust code on Windows, the difference being that it removes the dependency on Jenkins, so you don't need a Jenkins master running to spin it up.

It also updates the README to say we need to be using the development version of Ansible for now.

You can run `make travis_rust_slave-windows-2016-x86_64` to bring it up.

Cheers,

Chris

jacderida committed:
refactor/vm: more appropriate name

This machine needs a Jenkins master to run against, so it's better for
that to be reflected in the name. I'm going to create a similar machine
without the Jenkins slave installation.

jacderida committed:
feat/vm: windows slave replicating travis

This new slave removes any Jenkins depedency, so you can just spin up
the machine without needing a Jenkins setup.

S-Coyle reviewed (changes requested):
Currently `make clean` doesn't destroy the `travis_rust_slave-windows-2016-x86_64` - would be good to ensure it destroys all local VMs

S-Coyle commented:
@jacderida should this new VM be added to the VM table in the readme?

jacderida committed:
fix/clean: remove machines without error

Uses a little script to clean up all the local VMs. Unfortunately
Vagrant returns an error with a general `vagrant destroy -f` command if
any of the machines don't exist.

S-Coyle reviewed (changes requested):
Looks like it would make sense to update the readme's Windows Crust example to use the new `travis_rust_slave-windows-2016-x86_64` machine - https://github.com/maidsafe/safe-build-infrastructure/blob/da360cd8345fe17e9bbebbf0bc9c812c1c7c04af/README.md#windows

S-Coyle commented on scripts/sh/destroy_local_vms.sh:
```diff
@@ -0,0 +1,21 @@
+    "base-windows-2012_r2-x86_64" \
+    "rust_slave_git_bash-windows-2012_r2-x86_64" \
+    "jenkins_rust_slave-windows-2016-x86_64" \
+    "rust_slave_msys2-windows-2012_r2-x86_64")
```
Missed `travis_rust_slave-windows-2016-x86_64` from this list

jacderida replied:
Thanks for this. Pushed another commit, and it also contains the documentation change.

jacderida committed:
fix/script: add missing local machine

Also makes a slight change to the README based on review feedback. The
Windows machine to use for building Crust has been updated to the new
Travis slave.

jacderida committed:
fix/script: correct syntax error

jacderida committed:
docs/vm: describe the generally useful machines

There are a lot more VMs available in the Vagrantfile, but most of the
others have very specific purposes, for example a WireGuard sandbox, so
I don't think they're worth listing here.

I've also removed references to specific applications in the table,
because really these machines are quite generic and could be used for
building most Rust applications, rather than just SCL and Crust
specifically.

For much the same reason, I've also removed the section on
'Building Crust', because I don't think this repository should be
targetted to that specifically, and the instructions for building it
should probably just be contained in that repository, otherwise we'd
need to maintain 2 documents.

jacderida commented:
> Looks like it would make sense to update the readme's Windows Crust example to use the new `travis_rust_slave-windows-2016-x86_64` machine - https://github.com/maidsafe/safe-build-infrastructure/blob/da360cd8345fe17e9bbebbf0bc9c812c1c7c04af/README.md#windows

I've updated the table with the new machine. I also removed any references to specific products, e.g. "Use for building SCL", as really these machines can be used for building any Rust applications (although sometimes there might be some stuff specific to each product, but those would probably be part of a specific build slave).

I removed the whole section on 'Building Crust', because I don't think it makes much sense to have a reference to Crust here. We should probably just leave the documentation in their own repositories to describe how to build each specific codebase. To me it's fairly obvious that these machines can be used as generic utility machines.

S-Coyle commented:
Latest changes look good. Approved
